### DISCUSSION TITLE
I'm having a problem with createWebHistory

### DISCUSSION THREAD

**Discussion by ouiiuh** [vuejs/router#1832] [Help and Questions] [answered]
When I pass in the parameter createWebHistory('/a/b'), my url redirects from localhost/a to localhost/a/b/a, I actually need it to become localhost/a/b, how should I configure it

**Comment by posva** [comment, marked as answer]
You shouldn't worry about this: In practice, this feature allows your website to be hosted in any subdirectory of a website `example.com/app/` needs `/app` as a base. In this scenario, your users always enter your application through `example.com/app`, any other subdirectory will serve a different page without your application.

If what you want to add is a prefix to all routes, do that in the routes themselves:

```js
routes: [
  { path: '/app', children: [
    // actual routes
  ]}
]
```

**Reply by ouiiuh** [reply]
Ok, thank you very much for your reply
